src/service/errors/responses.go

```diff
@@ -1,7 +1,22 @@
 package errors
 
-import "github.com/hashicorp/vault/sdk/logical"
+import (
+	"github.com/ConsenSys/orchestrate-hashicorp-vault-plugin/src/pkg/errors"
+	"github.com/hashicorp/vault/sdk/logical"
+	"net/http"
+)
 
-func BadRequestError(req *logical.Request, msg string) (*logical.Response, error) {
-	return logical.RespondWithStatusCode(logical.ErrorResponse(msg), req, 400)
+func WriteHTTPError(req *logical.Request, err error) (*logical.Response, error) {
+	switch {
+	case errors.IsAlreadyExistsError(err):
+		return logical.RespondWithStatusCode(logical.ErrorResponse(err.Error()), req, http.StatusConflict)
+	case errors.IsNotFoundError(err):
+		return logical.RespondWithStatusCode(logical.ErrorResponse(err.Error()), req, http.StatusNotFound)
+	case errors.IsInvalidParameterError(err), errors.IsEncodingError(err):
+		return logical.RespondWithStatusCode(logical.ErrorResponse(err.Error()), req, http.StatusUnprocessableEntity)
+	case errors.IsInvalidFormatError(err):
+		return logical.RespondWithStatusCode(logical.ErrorResponse(err.Error()), req, http.StatusBadRequest)
+	default:
```

Comment:
in case err is nil will fall here even what looks incorrect

Reply:
I don't think there is a need to check if err is nil here. We already discussed this kind of topic but this is only to prevent developer error but normally err is never nil in this function.

Reply:
I know and I still believe we should code for future developers who might not know how to use the methods we implement xD